audio_helper/src/openxtalsa.c

```diff
@@ -0,0 +1,955 @@
+//
+// Copyright (c) 2015 Assured Information Security, Inc
+//
+// Dates Modified:
+//  - 4/8/2015: Initial commit
+//    Rian Quinn <quinnr@ainfosec.com>
+//
+// This program is free software; you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation; either version 2 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with this program; if not, write to the Free Software
+// Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
+//
+
+#include "openxtalsa.h"
+#include "openxtdebug.h"
+
+/////////////////////////////////////////////////////////////////////////////////////////////////////////
+// Global Variables                                                                                    //
+/////////////////////////////////////////////////////////////////////////////////////////////////////////
+
+static char device[256] = "default";
+
+/////////////////////////////////////////////////////////////////////////////////////////////////////////
+// Global Functions                                                                                    //
+/////////////////////////////////////////////////////////////////////////////////////////////////////////
+
+int openxt_alsa_set_card(int32_t n)
+{
+    int ret;
+
+    // Sanity checks
+    openxt_assert(n >= 0, -EINVAL);
+    openxt_assert(n < 32, -EINVAL);
+
+    // Copy the name to the global device name
+    ret = snprintf(device, 256, "hw:%i", n);
+    openxt_assert_ret(ret >= 0, ret, ret);
+
+    // Done
+    return 0;
+}
+
+int openxt_alsa_set_device(char *name)
+{
+    int ret;
+
+    // Sanity checks
+    openxt_checkp(name, -EINVAL);
+    openxt_assert(strlen(name) < 256, -EINVAL);
+
+    // Copy the name to the global device name
+    ret = snprintf(device, 256, "%s", name);
+    openxt_assert_ret(ret >= 0, ret, ret);
+
+    // Done
+    return 0;
+}
+
+/////////////////////////////////////////////////////////////////////////////////////////////////////////
+// Settings Functions                                                                                  //
+/////////////////////////////////////////////////////////////////////////////////////////////////////////
+
+int openxt_alsa_create(Settings **settings)
+{
+    // Sanity checks
+    openxt_checkp(settings, -EINVAL);
+    openxt_assert(*settings == NULL, -EINVAL);
+
+    // Allocate the settings variable.
+    *settings = (Settings *)malloc(sizeof(Settings));
+
+    // Make sure that the settings variable is cleared.
+    memset(*settings, 0, sizeof(Settings));
```

Comment:
If you need it memset 0 on something just malloc'd, perhaps just use calloc.

Also, you memset then you check the pointer returned from malloc... you should check pointer first.